### ISSUE TITLE
Unable to add content setting AllowedUploadFiles 

### ISSUE THREAD

**Issue by neilsalmondstorm** [umbraco/Umbraco-CMS#13308]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.3.1

### Bug summary

Hi,

I am trying to lock down the allowed upload file types to stop executables and other dangerous file types from being loaded to media in Umbraco 10. 
Using the guide [here](https://our.umbraco.com/Documentation/Reference/Configuration/ContentSettings/index-v9_1_0) I updated my appsettings and added 
"AllowedUploadFiles": [ "pdf", "doc", "docx" ...]
but the json validate says the value must be a string

When I change the value to a string
"AllowedUploadFiles": "pdf, doc, docx..."
the validator says the value at /Umbraco/CMS/Content/AllowedUploadFiles must be an array.

I checked appsettings-schema.json and see
AllowedUploadFiles has properties of string at line 858 and 
AllowedUploadFiles is type array at line 1800 

I took a local copy of appsettings-schema.json and set both to be array.
This fixed the json validator but didn't change the allowed upload file types in the backoffice.

I also added Azure appsetting 
Umbraco:CMS:Content:AllowedUploadFiles to [ "pdf", "doc", "docx" ...] to match [ContentSettings.cs](https://github.com/umbraco/Umbraco-CMS/blob/b69afe81f3f6fcd37480b3b0295a62af44ede245/src/Umbraco.Core/Configuration/Models/ContentSettings.cs#L204)
but I was still able to upload an exe.

Any help fixing this is appreciated.
Thanks,
Neil

### Steps to reproduce

Override Content Setting Umbraco:CMS:Content:AllowedUploadFiles to ["doc", "pdf"]  in appsettings.json

### Expected result / actual result

Expected Result
Only allow files with a suffix in the AllowedUploadFiles list to be uploaded to media
Actual results
Any file type except those defaulted in [DisallowedUploadFiles](https://github.com/umbraco/Umbraco-CMS/blob/b69afe81f3f6fcd37480b3b0295a62af44ede245/src/Umbraco.Core/Configuration/Models/ContentSettings.cs#L153) can be uploaded

**Comment by Ambertvu** [comment]
I've been working on this in the past to fix it, but lacking a bit of time, it seems that none of the CSV/Array entries are processed correctly. If I remember correctly, it's because the value is expected to be an IEnumerable, which a CSV value isn't.

**Comment by bergmania** [comment]
We are looking into this issue in this sprint, in the meantime, you can easily configure it using code, like this example:

```cs
public class WhateverComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        builder.Services.Configure<ContentSettings>(options =>
        {
            options.AllowedUploadFiles = new[] { "pdf" };
        });
    }
}
```

**Comment by neilsalmond** [comment]
Thanks @bergmania, works like a charm!

**Comment by nikolajlauridsen** [comment]
Fixed in #13408, thanks for reporting 😄

**Comment by neilsalmondstorm** [comment]
Thanks @nikolajlauridsen
